### PR TITLE
Conditional tag fixes #3

### LINKED ISSUE #3: Implement {conditional} tag
Implement the existing tag used in the CSFG. The function is found within [`markdownsection.py`](https://github.com/uccser/cs-field-guide/blob/develop/generator/markdownsection.py).

```
[conditional]
    regex: ^\{conditional-content ?(?P<args>[^\}]*)\}(?P<content>[\s\S]*?)\{conditional-content end\}
    function: check_conditional_content
```

This conditionally added content when creating files for static site generation. However the Django system should now perform the logic if this content should be displayed.

### PR DESCRIPTION
**Pull request by ravenmaster001**
Fixes #3 

This includes the requirements changes from the last PR I sent you @hayleyavw.

### CODE REVIEW

File: kordac/tests/assets/conditional/doc_example_override_html_expected.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<script>
+ if (version == 'teacher') {
+  return "<p>This is text that only teachers should see.</p>
+";
```
Comment (hayleyavw) [inline kordac/tests/assets/conditional/doc_example_override_html_expected.html]: why are the end quotes on a newline, is this intentional? Github's highlighting of this burns my eyes

File: kordac/tests/assets/smoke/algorithms.md [hunk 2]
```diff
@@ -110,7 +110,7 @@ Not all algorithms take double the time for double the input; some take a lot mo
 That's worth knowing in advance because we usually need our programs to scale up well;
 in the case of the high scores, if you're running a game that suddenly becomes popular, you want to know in advance that the high score algorithm will be fast enough if you get more scores to check.
 
-{panel type="extra-for-experts" "Algorithm complexity"}
+{panel type="extra-for-experts" title="Extra For Experts" title="Extra For Experts" "Algorithm complexity"}
```
Comment (hayleyavw) [inline kordac/tests/assets/smoke/algorithms.md]: ahhh....thought you fixed this? why are there two titles again?

File: kordac/tests/assets/smoke/algorithms.md [hunk 3]
```diff
@@ -137,7 +137,7 @@ They also happen to be great for illustrating some of the key concepts that aris
 
 ## Searching
 
-{panel type="teacher-note" summary="Presenting searching in the classroom"}
+{panel type="teacher-note" title="Teacher Note" title="teacher-note" summary="Presenting searching in the classroom"}
```
Comment (hayleyavw) [inline kordac/tests/assets/smoke/algorithms.md]: and again....

**Comment by JackMorganNZ** [comment]
This PR is missing two pieces of documentation:

1. Section on overriding HTML (see other processors - provide *simple* example)
2. A warning that this processor can't be nested (Unless it can?? But I think we should force one layer only)

File: kordac/tests/assets/smoke/algorithms.md [hunk 4]
```diff
@@ -110,7 +110,7 @@ Not all algorithms take double the time for double the input; some take a lot mo
 That's worth knowing in advance because we usually need our programs to scale up well;
 in the case of the high scores, if you're running a game that suddenly becomes popular, you want to know in advance that the high score algorithm will be fast enough if you get more scores to check.
 
-{panel type="extra-for-experts" title="Extra For Experts" title="Extra For Experts" "Algorithm complexity"}
+{panel type="extra-for-experts" title="Extra For Experts" "Algorithm complexity"}
```
Comment (hayleyavw) [inline kordac/tests/assets/smoke/algorithms.md]: remove arbitrary text

**Pull request merged**